### PR TITLE
mgmt: smp: Fix NULL pointer dereferences in UDP transport

### PR DESCRIPTION
**Pull request by berendo**
This PR fixes two `NULL` pointer dereferences in the SMP UDP transport. The first one occurs if the network manager sends `NET_EVENT_L4_DISCONNECTED` _before_ `NET_EVENT_L4_CONNECTED` (because, for example, another service has called `net_conn_mgr_resend_status()` during its initialization), which leads to `smp_udp_close()` attempting to `k_thread_abort()` an as-of-yet-uninitialized thread. The second one is due to an improper use of `inet_pton()` where `INADDR_ANY` is passed as if it's a dotted-decimal string, which leads to a `strlen()` call on `0`.

### CODE REVIEW

File: subsys/mgmt/mcumgr/smp_udp.c [hunk 1]
```diff
@@ -204,7 +206,7 @@ int smp_udp_open(void)
 	memset(&addr4, 0, sizeof(addr4));
 	addr4.sin_family = AF_INET;
 	addr4.sin_port = htons(CONFIG_MCUMGR_SMP_UDP_PORT);
-	inet_pton(AF_INET, INADDR_ANY, &addr4.sin_addr);
+	addr4.sin_addr.s_addr = htonl(INADDR_ANY);
```
Comment (de-nordic) [inline subsys/mgmt/mcumgr/smp_udp.c:209]: The change does not seem related to claimed fix, should this receive separate commit in this PR?

File: subsys/mgmt/mcumgr/smp_udp.c [hunk 2]
```diff
@@ -240,13 +242,19 @@ int smp_udp_open(void)
...
+		k_thread_abort(&(configs.ipv4.thread));
+		close(configs.ipv4.sock);
+		configs.ipv4.sock = -1;
+	}
 #endif
 
 #if CONFIG_MCUMGR_SMP_UDP_IPV6
-	k_thread_abort(&(configs.ipv6.thread));
-	close(configs.ipv6.sock);
+	if (configs.ipv4.sock >= 0) {
```
Comment (de-nordic) [inline subsys/mgmt/mcumgr/smp_udp.c:253]: Shouldn't this be ipv6?

File: subsys/mgmt/mcumgr/smp_udp.c [hunk 3]
```diff
@@ -240,13 +242,19 @@ int smp_udp_open(void)
...
+		k_thread_abort(&(configs.ipv6.thread));
+		close(configs.ipv6.sock);
+		configs.ipv4.sock = -1;
```
Comment (de-nordic) [inline subsys/mgmt/mcumgr/smp_udp.c:256]: Shouldn't this be ipv6?

Comment (berendo) [reply subsys/mgmt/mcumgr/smp_udp.c]: Yes. Copy/paste 🤦‍♂️... Fixed and force-pushed.

File: subsys/mgmt/mcumgr/smp_udp.c (hunk 1 above)
Comment (berendo) [reply subsys/mgmt/mcumgr/smp_udp.c:209]: Hmmm... It _is_ a `NULL`-pointer dereference that is being fixed. `INADDR_ANY` (which is `0`) was being passed in place of a `char *` argument in `inet_pton()`, which resulted in a `NULL`-pointer dereference in `strlen()` (though the code would "silently" work in most systems which have a `0` at or near address `0`). While the fix itself may change the "API" being used, it is nevertheless in the spirit of the `NULL`-pointer dereference fixes that the PR describes.

Nevertheless, if you think it absolutely must be split out, please let me know and I can split the PR into two _really_ tiny ones.

Comment (de-nordic) [reply subsys/mgmt/mcumgr/smp_udp.c:209]: OK, my bad.

**Comment by de-nordic** [comment]
@aunsbjerg Can you also take a look?

**Review by de-nordic — APPROVED** [review]
Looks OK.

**Review by aunsbjerg — APPROVED** [review]
LGTM

**Comment by de-nordic** [comment]
> LGTM

Thanks.

**Review by nandojve — APPROVED** [review]

File: subsys/mgmt/mcumgr/smp_udp.c (hunk 1 above)
Comment (berendo) [reply subsys/mgmt/mcumgr/smp_udp.c:209]: No worries and thank you for the review.
